tests/store/artifact/test_databricks_artifact_repo.py

```diff
@@ -274,6 +275,132 @@ def test_log_artifact_azure_blob_client_sas_error(self, databricks_artifact_repo
+        }
+        filtered_azure_headers = {
+            "x-ms-content-type": "test-type",
+            "x-ms-owner": "some-owner",
+        }
+        mock_response = Response()
+        mock_response.status_code = 200
+        mock_response.close = lambda: None
+        with mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY + "._get_write_credential_infos"
+        ) as write_credential_infos_mock, mock.patch(
+            "mlflow.utils.rest_utils.cloud_storage_http_request"
+        ) as request_mock, mock.patch(
+            "mlflow.store.artifact.databricks_artifact_repo._AZURE_MAX_BLOCK_CHUNK_SIZE",
+            5,
+        ):
+            mock_credential_info = ArtifactCredentialInfo(
+                signed_uri=MOCK_ADLS_GEN2_SIGNED_URI,
+                type=ArtifactCredentialType.AZURE_ADLS_GEN2_SAS_URI,
+                headers=[
+                    ArtifactCredentialInfo.HttpHeader(name=header_name, value=header_value)
+                    for header_name, header_value in mock_azure_headers.items()
+                ],
+            )
+            write_credential_infos_mock.return_value = [mock_credential_info]
+            request_mock.return_value = mock_response
+            databricks_artifact_repo.log_artifact(test_file.strpath, artifact_path)
+            write_credential_infos_mock.assert_called_with(
+                run_id=MOCK_RUN_ID, paths=[expected_location]
+            )
+            # test with block size 5
+            request_mock.assert_any_call(
+                "put",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?resource=file",
+                headers=filtered_azure_headers,
+            )
+            request_mock.assert_any_call(
+                "patch",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?action=append&position=0",
+                data=ANY,
+                headers=filtered_azure_headers,
+            )
+            request_mock.assert_any_call(
+                "patch",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?action=append&position=5",
+                data=ANY,
+                headers=filtered_azure_headers,
+            )
+            request_mock.assert_any_call(
+                "patch",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?action=append&position=10",
+                data=ANY,
+                headers=filtered_azure_headers,
+            )
+            request_mock.assert_called_with(
+                "patch",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?action=flush&position=14",
+                headers=filtered_azure_headers,
+            )
+
+    def test_log_artifact_adls_gen2_flush_error(self, databricks_artifact_repo, test_file):
+        mock_successful_response = Response()
+        mock_successful_response.status_code = 200
+        mock_successful_response.close = lambda: None
+        mock_error_response = MlflowException("MOCK ERROR")
+        with mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY + "._get_write_credential_infos"
```

Comment:
```suggestion
            DATABRICKS_ARTIFACT_REPOSITORY + "._get_write_credential_infos",
            return_value=[...]
```

Nit: we can specify `return_value` when calling `mock.patch`

Reply:
Fixing